sdk/tables/azure-data-tables/src/main/java/com/azure/data/tables/TablesSharedKeyCredential.java

```diff
@@ -0,0 +1,111 @@
+        this.accountName = accountName;
+        this.accountKey = accountKey;
+    }
+
+    /**
+     * Generates the Auth Headers
+     *
+     * @param requestURL the URL which the request is going to
+     * @param headers the headers of the request
+     * @return the auth header
+     */
+    public String generateAuthorizationHeader(URL requestURL, Map<String, String> headers) {
+        String signature = StorageImplUtils.computeHMac256(this.accountKey, this.buildStringToSign(requestURL,
+            headers));
+        return String.format(AUTHORIZATION_HEADER_FORMAT, this.accountName, signature);
+    }
+
+    /**
+     * creates the String to Sign
+     *
+     * @param requestURL the URL which the request is going to
+     * @param headers the headers of the request
+     * @return a string to sign for the request
+     */
+    private String buildStringToSign(URL requestURL, Map<String, String> headers) {
+        String dateHeader = headers.containsKey("x-ms-date") ? "" : this.getStandardHeaderValue(headers,
+            "Date");
+        return String.join("\n",
+            dateHeader,  //date
+            this.getCanonicalizedResource(requestURL)); //Canonicalized resource
+    }
+
+    /**
+     * gets necessary headers if the request does not already contain them
+     *
+     * @param headers a map of the headers which the request has
+     * @param headerName the name of the header to get the standard header for
+     * @return the standard header for the given name
+     */
+    private String getStandardHeaderValue(Map<String, String> headers, String headerName) {
+        String headerValue = headers.get(headerName);
+        return headerValue == null ? "" : headerValue;
+    }
+
+
+    /**
+     * returns the canonicalized resource needed for a request
+     *
+     * @param requestURL the url of the request
+     * @return the string that is the canonicalized resource
+     */
+    private String getCanonicalizedResource(URL requestURL) {
+        StringBuilder canonicalizedResource = new StringBuilder("/");
+        canonicalizedResource.append(this.accountName);
+        if (requestURL.getPath().length() > 0) {
+            canonicalizedResource.append(requestURL.getPath());
+        } else {
+            canonicalizedResource.append('/');
+        }
+
+        if (requestURL.getQuery() != null) {
+            Map<String, String[]> queryParams = StorageImplUtils.parseQueryStringSplitValues(requestURL.getQuery());
+            ArrayList<String> queryParamNames = new ArrayList<>(queryParams.keySet());
+            Collections.sort(queryParamNames);
+
+            for (String queryParamName : queryParamNames) {
+                String[] queryParamValues = queryParams.get(queryParamName);
+                Arrays.sort(queryParamValues);
+                String queryParamValuesStr = String.join(",", queryParamValues);
+                if (queryParamName.equals("comp")) {
```

Comment:
My personal rule of thumb is to add an empty line between all my variable creations and method invocations. I think it makes it easier to reason about as you're reading. In this case, it would look like:

```java
Map<String, String[]> queryParams = StorageImplUtils.parseQueryStringSplitValues(requestURL.getQuery());
ArrayList<String> queryParamNames = new ArrayList<>(queryParams.keySet());

Collections.sort(queryParamNames);

if (requestURL.getQuery() != null) {
    for (String queryParamName : queryParamNames) {
        String[] queryParamValues = queryParams.get(queryParamName);

        Arrays.sort(queryParamValues);

        String queryParamValuesStr = String.join(",", queryParamValues);

        if (queryParamName.equals("comp")) {
            canonicalizedResource.append("?").append(queryParamName.toLowerCase(Locale.ROOT))
                .append("=")
                .append(queryParamValuesStr);
        }
    }
}

return canonicalizedResource.toString();
```